keanu-project/src/test/java/io/improbable/keanu/e2e/regression/RadonHierarchicalRegression.java

```diff
@@ -0,0 +1,183 @@
+package io.improbable.keanu.e2e.regression;
+
+import io.improbable.keanu.DeterministicRule;
+import io.improbable.keanu.algorithms.variational.optimizer.gradient.GradientOptimizer;
+import io.improbable.keanu.model.regression.RegressionModel;
+import io.improbable.keanu.model.regression.RegressionRegularization;
+import io.improbable.keanu.network.BayesianNetwork;
+import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.util.csv.ReadCsv;
+import io.improbable.keanu.vertices.VertexLabel;
+import io.improbable.keanu.vertices.dbl.DoubleVertex;
+import io.improbable.keanu.vertices.dbl.probabilistic.GaussianVertex;
+import io.improbable.keanu.vertices.dbl.probabilistic.HalfGaussianVertex;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.both;
+import static org.hamcrest.Matchers.greaterThan;
+import static org.hamcrest.Matchers.lessThan;
+
+public class RadonHierarchicalRegression {
+
+    @Rule
+    public DeterministicRule rule = new DeterministicRule();
+
+    private List<Data> radonData;
+
+    @Before
+    public void readRadonCSV() {
+        radonData = ReadCsv.fromResources("data/datasets/radon/radon.csv")
+            .asRowsDefinedBy(Data.class)
+            .load(true);
+    }
+
+    @Test
+    public void canPerformSimpleLinearRegression() {
+        RegressionModel model = linearRegression(radonData);
+        assertThat(model.getWeightVertex().getValue().scalar(), both(greaterThan(-0.7)).and(lessThan(-0.4)));
+        assertThat(model.getInterceptVertex().getValue().scalar(), both(greaterThan(1.2)).and(lessThan(1.5)));
+    }
+
+    @Test
+    public void canPerformRegressionWithOneHierarchy() {
+        buildAndRunHeirarchicalNetwork(radonData, 1);
+    }
+
+    @Test
+    public void canPerformRegressionWithTwoHierarchies() {
+        buildAndRunHeirarchicalNetwork(radonData, 2);
+    }
+
+    @Test
+    public void canPerformRegressionWithFourHierarchies() {
+        buildAndRunHeirarchicalNetwork(radonData, 4);
+    }
+
+    @Test
+    public void canPerformRegressionWithTenHierarchies() {
+        buildAndRunHeirarchicalNetwork(radonData, 10);
```

Comment:
You'd mentioned before that this was there until we had the sampling fitter.  Now we have it - can we use it instead of doing this arbitrary split?